packages/contracts/test/chugsplash/deploy.spec.ts

```diff
@@ -35,22 +37,65 @@ describe('ChugSplash deploy script', () => {
         require(CONFIG_PATH)
       )
     })
+
     it('should correctly send executeAction transactions', async () => {
       await L2ChugSplashDeployer.connect(signer).approveTransactionBundle(
         currActionBundle.root,
         currActionBundle.actions.length
       )
 
-      const receipts = await executeActionsFromConfig(
+      const receipts = await executeActionsFromConfig({
+        hre,
+        signer,
+        chugsplashDeployerAddress: L2ChugSplashDeployer.address,
+        upgradeConfigPath: CONFIG_PATH,
+      })
+
+      expect(receipts.length).to.eq(currActionBundle.actions.length)
+    })
+
+    it('should not execute if the bundle is not approved', async () => {
+      const receipts = await executeActionsFromConfig({
+        hre,
+        signer,
+        chugsplashDeployerAddress: L2ChugSplashDeployer.address,
+        upgradeConfigPath: CONFIG_PATH,
+        timeoutInMs: 2_000,
+      })
+
+      expect(receipts).to.be.undefined
+    })
+
+    it('should keep retrying until timeout', async () => {
+      const receipts = await executeActionsFromConfig({
         hre,
         signer,
-        L2ChugSplashDeployer.address,
-        CONFIG_PATH
+        chugsplashDeployerAddress: L2ChugSplashDeployer.address,
+        upgradeConfigPath: CONFIG_PATH,
+      })
+
+      await L2ChugSplashDeployer.connect(signer).approveTransactionBundle(
+        currActionBundle.root,
+        currActionBundle.actions.length
       )
 
+      // setTimeout(async () => {await L2ChugSplashDeployer.connect(signer).approveTransactionBundle(
+      //   currActionBundle.root,
+      //   currActionBundle.actions.length
+      // )}, 2_000)
+
       expect(receipts.length).to.eq(currActionBundle.actions.length)
     })
```

Comment:
this test is currently failing but i'm not completely sure if it SHOULD succeed!